File: examples/openai-agents-sdk/verify.mjs
/**
 * Proves the Tenki-facing part of this example: an OpenAI Agents SDK tool that
 * executes Python in a live Tenki sandbox. No LLM needed — we call the tool
 * directly, exactly as the Agents runner would (a RunContext + a JSON string of
 * arguments), and assert its output. (The full agent, which needs a model key,
 * is in agent.mjs.)
 * Token/project from env (CI) or ~/.config/tenki/config.yaml (local `tenki login`).
 */
import { RunContext } from "@openai/agents";
import { TenkiSandbox } from "@tenkicloud/sandbox";
import { readFileSync } from "node:fs";
import { homedir } from "node:os";
import { makeCodeTool } from "./tenki-tool.mjs";

const cfg = (key) => {
	try {
		const c = readFileSync(`${homedir()}/.config/tenki/config.yaml`, "utf8");
		return (c.match(new RegExp(`^${key}:\\s*(.+)$`, "m"))?.[1] ?? "").trim();
	} catch {
		return "";
	}
};

const authToken = process.env.TENKI_AUTH_TOKEN || process.env.TENKI_API_KEY || cfg("auth_token");
const projectId = process.env.TENKI_PROJECT_ID || cfg("current_project_id");
const workspaceId = process.env.TENKI_WORKSPACE_ID || cfg("current_workspace_id");
if (!authToken) {
	console.error("No token. Set TENKI_AUTH_TOKEN, or run `tenki login`.");
	process.exit(1);
}

const tenki = new TenkiSandbox({ authToken });
let sandbox;
try {
	sandbox = await tenki.createAndWait({ cpuCores: 1, memoryMb: 1024, projectId, workspaceId });
	const runPython = makeCodeTool(sandbox);

	// Drive the Agents SDK function tool exactly as the runner would.
	const out = await runPython.invoke(new RunContext(), JSON.stringify({ code: "print(sum(range(11)))" }));
	if (String(out).trim() !== "55") throw new Error(`tool returned ${JSON.stringify(out)}`);

	console.log("✓ openai-agents-sdk: OpenAI Agents SDK tool executed Python in a Tenki sandbox → 55");
} catch (e) {
	console.error("✗ " + (e?.message ?? e));
	process.exitCode = 1;
} finally {
	if (sandbox) {
		try {
			await sandbox[Symbol.asyncDispose]();
		} catch {
			/* self-reaps via idle/lifetime caps */
		}
	}
}

File: examples/crewai-code-interpreter/verify.py
"""
Proves the Tenki-facing half of this example: the CrewAI tool executes Python in
a live Tenki sandbox. No LLM needed — we call the tool directly and assert its
output. (The full crew, which needs a model key, is in agent.py.)

Needs Python 3.10+ and the deps in requirements.txt. Token/project from env (CI)
or ~/.config/tenki/config.yaml (local `tenki login`).
"""
import os
import sys

from tenki_sandbox import Sandbox
from tenki_tool import TenkiCodeInterpreter


def cfg(key):
    try:
        with open(os.path.expanduser("~/.config/tenki/config.yaml")) as f:
            for line in f:
                if line.startswith(key + ":"):
                    return line.split(":", 1)[1].strip()
    except Exception:
        pass
    return ""


token = os.environ.get("TENKI_AUTH_TOKEN") or os.environ.get("TENKI_API_KEY") or cfg("auth_token")
if not token:
    print("No token. Set TENKI_AUTH_TOKEN, or run `tenki login`.")
    sys.exit(1)

# The Python SDK (unlike the Node SDK) doesn't auto-handle a bare browser session
# token — it sends anything without a known prefix as `Authorization: Bearer`,
# which the server rejects. A `tk_` API key works as-is; a `tenki login` session
# token must be sent as a cookie, which the SDK does when you prefix it `cookie:`.
# (SDK auth gap — see the pip-tenki / Python-SDK-auth issue.)
if not token.startswith(("tk_", "ory_st_", "cookie:")):
    token = f"cookie:{token}"

opts = {"auth_token": token, "cpu_cores": 1, "memory_mb": 1024}
project_id = os.environ.get("TENKI_PROJECT_ID") or cfg("current_project_id")
workspace_id = os.environ.get("TENKI_WORKSPACE_ID") or cfg("current_workspace_id")
if project_id:
    opts["project_id"] = project_id
if workspace_id:
    opts["workspace_id"] = workspace_id

try:
    with Sandbox.create(**opts) as sb:
        tool = TenkiCodeInterpreter(sandbox=sb)
        # Drive the CrewAI tool exactly as an agent would.
        out = tool._run(code="print(sum(range(11)))")
        if str(out).strip() != "55":
            raise AssertionError(f"tool returned {out!r}")
    print("✓ crewai-code-interpreter: CrewAI tool executed Python in a Tenki sandbox → 55")
except Exception as e:  # noqa
    print(f"✗ {type(e).__name__}: {e}")
    sys.exit(1)
